src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+            list.Add(item);
+            list.Remove(item);
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void AddRemoveDifferentValueEmptyCollection()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(0);
+            var list = GetList(items);
+            var item = GenerateItem();
+            list.Add(item);
+            items = items.Push(item);
+            list.Remove(GenerateItem());
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void AddNonNullRemoveNullEmptyCollection()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(0);
+            var list = GetList(items);
+            var item = GenerateItem();
+            list.Add(item);
+            items = items.Push(item);
+            if (ItemsMustBeNonNull && SearchingThrowsFromInvalidValue)
+            {
+                Assert.Throws<ArgumentException>(
+                    () => list.Remove(null));
+            }
+            else
+            {
+                list.Remove(null);
+            }
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void AddRemoveNullEmptyCollection()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            if (ItemsMustBeNonNull)
+                return;
+            var items = GenerateItems(0);
+            var list = GetList(items);
+            list.Add(null);
+            list.Remove(null);
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void AddNullRemoveNonNullEmptyCollection()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            if (ItemsMustBeNonNull)
+                return;
+            var items = GenerateItems(0);
+            var list = GetList(items);
+            list.Add(null);
+            items = items.Push(null);
+            list.Remove(GenerateItem());
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void RemoveItemsExistsTwice()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            if (ItemsMustBeUnique)
+                return;
+
+            var items = GenerateItems(16);
+            var tempItems = (object[]) items.Clone();
+            var list = GetList(items);
+            list.AddRange(items);
+            for (int i = 0; i < items.Length; i++)
+            {
+                int index = i + 1;
+                items =
+                    new object[
+                        tempItems.Length + tempItems.Length - index];
+                Array.Copy(tempItems, index, items, 0, 16 - index);
+                Array.Copy(tempItems, 0, items, 16 - index, 16);
+
```

Comment:
Actually, I won't change this. Part of the test is making sure that the correct item is removed. Without the tempItems array we can't validate that.